### PR TITLE
[SPARK-48129][PYTHON] Provide a constant table schema in PySpark for querying structured logs

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Similar to https://github.com/apache/spark/pull/46375/, this PR provides a constant table schema in PySpark for querying structured logs.
The doc of logging configuration is also updated.

### Why are the changes needed?

Provide a convenient way to query Spark logs using PySpark.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manual test

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: python/pyspark/util.py [hunk 1]
```diff
@@ -107,6 +107,22 @@ def majorMinorVersion(sparkVersion: str) -> Tuple[int, int]:
             )
 
 
+class LogUtils:
```
Comment (gengliangwang) [inline python/pyspark/util.py:110]: @HyukjinKwon Is there a better file location to put this constant string? There won't be a lot of new code in LogUtils.

Comment (HyukjinKwon) [reply python/pyspark/util.py:110]: Yeah, I think so

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: python/pyspark/util.py (hunk 1 above)
Comment (gengliangwang) [reply python/pyspark/util.py:110]: @HyukjinKwon where should it be?

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Let me merge this for Apache spark 4.0.0-preview.
We can revise the location later before 4.0.0 (as a follow-up PR).

**Comment by gengliangwang** [comment]
Thank you, @dongjoon-hyun @HyukjinKwon

**Pull request closed without merging**